em_examples/HarmonicVMDCylWidget.py

```diff
@@ -181,139 +246,193 @@ def getData(self, bType="b"):
-                ax.set_yticks([])
-                return "Dude, think twice ... only Jy for VMD"
-
-        out = Utils.plot2Ddata(
-            self.mesh2D.gridCC, val, vec=vec, ax=ax,
-            contourOpts={"cmap": "viridis"}, ncontour=50, scale=scale
-        )
-        if scale == "linear":
-            cb = plt.colorbar(
-                out[0], ax=ax, ticks=np.linspace(val.min(), val.max(), 3),
-                format="%.1e"
-            )
-        elif scale == "log":
-            cb = plt.colorbar(
-                out[0], ax=ax, ticks=np.linspace(val.min(), val.max(), 3),
-                format="$10^{%.1f}$"
+            elif Field == "E":
+                label = "Electric field (V/m)"
+                if view == "y":
+                    if ComplexNumber == "real":
+                        val = self.Ey.real
+                    elif ComplexNumber == "imag":
+                        val = self.Ey.imag
+                    elif ComplexNumber == "amplitude":
+                        val = abs(self.Ey)
+                    elif ComplexNumber == "phase":
+                        val = np.angle(self.Ey)
+                else:
+                    # ax.imshow(self.im)
+                    ax.set_xticks([])
+                    ax.set_yticks([])
+                    print("Think about the problem geometry. There is NO Ex or Ez in this case.")
+
+            elif Field == "J":
+                label = "Current density (A/m$^2$)"
+                if view == "y":
+                    if ComplexNumber == "real":
+                        val = self.Jy.real
+                    elif ComplexNumber == "imag":
+                        val = self.Jy.imag
+                    elif ComplexNumber == "amplitude":
+                        val = abs(self.Jy)
+                    elif ComplexNumber == "phase":
+                        val = np.angle(self.Jy)
+                else:
+                    # ax.imshow(self.im)
+                    ax.set_xticks([])
+                    ax.set_yticks([])
+                    print("Think about the problem geometry. There is NO Jx or Jz in this case.")
+
+            out = Utils.plot2Ddata(
+                self.mesh2D.gridCC, val, vec=vec, ax=ax,
+                contourOpts={"cmap": "viridis"}, ncontour=200, scale=scale
             )
-        else:
-            raise Exception("We consdier only linear and log scale!")
-        cb.set_label(label)
-        xmax = self.mesh2D.gridCC[:, 0].max()
-        if Geometry:
-            ax.plot(
-                np.r_[-xmax, xmax], np.ones(2)*self.srcLoc[2], 'k-', lw=0.5
+            if scale == "linear":
+                cb = plt.colorbar(
+                    out[0], ax=ax, ticks=np.linspace(val.min(), val.max(), 3),
+                    format="%.1e"
+                )
+            elif scale == "log":
+                cb = plt.colorbar(
+                    out[0], ax=ax, ticks=np.linspace(val.min(), val.max(), 3),
+                    format="$10^{%.1f}$"
```

Comment:
These ticks aren't showing up, but one thing you could do is
```python
cb = plt.colorbar(
    out[0], ax=ax, format="$10^{%.0f}$"
)
ticks = cb.get_ticks()
cb.set_ticks(ticks[[0, -1]])  # grabs the first and last tick that are displayed
```

and it will show up as 

![image](https://user-images.githubusercontent.com/6361812/42847097-c129942e-89cf-11e8-8eab-36e92c354f98.png)

Reply:
@sgkang: you might also have some advice on how to set a good colorbar for log scales?